homeassistant/components/snooz/fan.py

```diff
@@ -0,0 +1,155 @@
+"""Fan representation of a Snooz device."""
+from __future__ import annotations
+
+from collections.abc import Callable
+import logging
+from typing import Any
+
+from pysnooz.api import SnoozDeviceState, UnknownSnoozState
+from pysnooz.commands import (
+    SnoozCommandData,
+    SnoozCommandResultStatus,
+    set_volume,
+    turn_off,
+    turn_on,
+)
+from pysnooz.device import SnoozConnectionStatus, SnoozDevice
+
+from homeassistant.components.fan import ATTR_PERCENTAGE, FanEntity, FanEntityFeature
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_ADDRESS, STATE_OFF, STATE_ON
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.restore_state import RestoreEntity
+
+from .const import DOMAIN
+from .models import SnoozConfigurationData
+
+# transitions logging is pretty verbose, so only enable warnings/errors
+logging.getLogger("transitions.core").setLevel(logging.WARNING)
+
```

Comment:
```suggestion
```

This affects the global state so we don't want to change it in an integration.

Can the library be fixed instead?

Reply:
Are you suggesting making [logs like these](https://github.com/pytransitions/transitions/blob/56949457a4b2e6e2e932a939ec1dddb29ba0f302/transitions/core.py#L130) debug level in the library?

Reply:
I'm suggesting this is a bit of hack to do this since one integration is affecting the global state.

The library shouldn't be logging at this level in the first place.  I'd probably pick a different library but its probably a little late to do that 😉 so if there isn't a better solution its ok to leave as-is.

Reply:
I agree that we shouldn't be affecting the global state of the logger from within an entity file.  Is there any way to enrich the default logger component settings in the build system?

Something to the effect of
```yaml
transitions.core: warning
```

Reply:
We do this in bootstrap https://github.com/home-assistant/core/blob/89d0b434bcf0f0bd42eb055b575da77449de38bf/homeassistant/bootstrap.py#L335 but only for base requirements so there isn't really a place you could add it since the library appears to only be used by this integration (or at least I have nothing else installed in my dev env that uses it according to `pipdeptree`)

I think we are stuck with this.

```
pysnooz==0.7.8
  - bleak [required: >=0.18.1, installed: 0.18.1]
    - async-timeout [required: >=3.0.0,<5, installed: 4.0.2]
    - pyobjc-core [required: >=8.5,<9.0, installed: 8.5]
    - pyobjc-framework-CoreBluetooth [required: >=8.5,<9.0, installed: 8.5]
      - pyobjc-core [required: >=8.5, installed: 8.5]
      - pyobjc-framework-Cocoa [required: >=8.5, installed: 8.5]
        - pyobjc-core [required: >=8.5, installed: 8.5]
    - pyobjc-framework-libdispatch [required: >=8.5,<9.0, installed: 8.5]
      - pyobjc-core [required: >=8.5, installed: 8.5]
  - bleak-retry-connector [required: >=2.1.3, installed: 2.1.3]
    - async-timeout [required: >=4.0.1, installed: 4.0.2]
    - bleak [required: >=0.18.0, installed: 0.18.1]
      - async-timeout [required: >=3.0.0,<5, installed: 4.0.2]
      - pyobjc-core [required: >=8.5,<9.0, installed: 8.5]
      - pyobjc-framework-CoreBluetooth [required: >=8.5,<9.0, installed: 8.5]
        - pyobjc-core [required: >=8.5, installed: 8.5]
        - pyobjc-framework-Cocoa [required: >=8.5, installed: 8.5]
          - pyobjc-core [required: >=8.5, installed: 8.5]
      - pyobjc-framework-libdispatch [required: >=8.5,<9.0, installed: 8.5]
        - pyobjc-core [required: >=8.5, installed: 8.5]
  - bluetooth-sensor-state-data [required: >=1.5.0, installed: 1.6.0]
    - home-assistant-bluetooth [required: >=1.3.0, installed: 1.3.0]
      - bleak [required: >=0.14.3, installed: 0.18.1]
        - async-timeout [required: >=3.0.0,<5, installed: 4.0.2]
        - pyobjc-core [required: >=8.5,<9.0, installed: 8.5]
        - pyobjc-framework-CoreBluetooth [required: >=8.5,<9.0, installed: 8.5]
          - pyobjc-core [required: >=8.5, installed: 8.5]
          - pyobjc-framework-Cocoa [required: >=8.5, installed: 8.5]
            - pyobjc-core [required: >=8.5, installed: 8.5]
        - pyobjc-framework-libdispatch [required: >=8.5,<9.0, installed: 8.5]
          - pyobjc-core [required: >=8.5, installed: 8.5]
    - sensor-state-data [required: >=2.0, installed: 2.7.0]
  - Events [required: >=0.4,<0.5, installed: 0.4]
  - home-assistant-bluetooth [required: >=1.3.0, installed: 1.3.0]
    - bleak [required: >=0.14.3, installed: 0.18.1]
      - async-timeout [required: >=3.0.0,<5, installed: 4.0.2]
      - pyobjc-core [required: >=8.5,<9.0, installed: 8.5]
      - pyobjc-framework-CoreBluetooth [required: >=8.5,<9.0, installed: 8.5]
        - pyobjc-core [required: >=8.5, installed: 8.5]
        - pyobjc-framework-Cocoa [required: >=8.5, installed: 8.5]
          - pyobjc-core [required: >=8.5, installed: 8.5]
      - pyobjc-framework-libdispatch [required: >=8.5,<9.0, installed: 8.5]
        - pyobjc-core [required: >=8.5, installed: 8.5]
  - transitions [required: >=0.8.11,<0.9.0, installed: 0.8.11]
    - six [required: Any, installed: 1.16.0]
```

Reply:
Let's move it to `async_setup_entry` in `__init__.py` and add a comment

Reply:
I'll also see if I can submit a PR to make the `info` logs become `debug` by default since I think it's a reasonable ask for this type of scenario.

Reply:
After some digging it seems like this was already brought up - https://github.com/pytransitions/transitions/pull/62.